### PR TITLE
Banning shared_ptr and std::cerr

### PR DESCRIPTION
**Pull request by lmwnshn**
Remove or NOLINT usages of shared_ptr and std::cerr from the codebase.

### CODE REVIEW

**Comment by lmwnshn** [comment]
Sure, I have no strong opinions.

**Comment by lmwnshn** [comment]
Blocked on logger, please see slack.

**Comment by mbutrovich** [comment]
We can’t merge this until all of the refactors are done.

File: src/include/network/network_io_wrapper.h [hunk 1]
```diff
@@ -94,20 +94,20 @@ class NetworkIoWrapper {
...
    */
-  std::shared_ptr<WriteQueue> GetWriteQueue() { return out_; }
+  std::shared_ptr<WriteQueue> GetWriteQueue() { return out_; }  // NOLINT
 
  private:
   // The file descriptor associated with this NetworkIoWrapper
   int sock_fd_;
   // The ReadBuffer associated with this NetworkIoWrapper
-  std::shared_ptr<ReadBuffer> in_;
+  std::shared_ptr<ReadBuffer> in_;  // NOLINT
```
Comment (lmwnshn) [inline src/include/network/network_io_wrapper.h]: So I'm not entirely sure who owns what when. Ditto for out_. Can these be managed pointers? See other comment.

File: src/include/optimizer/operator_node.h [hunk 2]
```diff
@@ -202,7 +202,7 @@ class Operator {
     if (node_) {
       auto &n = *node_;
       if (typeid(n) == typeid(T)) {
-        return reinterpret_cast<const T *>(node_.get());
+        return reinterpret_cast<const T *>(node_.Get());
```
Comment (lmwnshn) [inline src/include/optimizer/operator_node.h]: Someone, possibly me, sanity-check that this still makes sense to do.

File: src/include/optimizer/operator_node.h [hunk 3]
```diff
@@ -212,7 +212,7 @@ class Operator {
   /**
    * Pointer to the base operator
    */
-  std::shared_ptr<BaseOperatorNode> node_;
+  common::ManagedPointer<BaseOperatorNode> node_;
```
Comment (lmwnshn) [inline src/include/optimizer/operator_node.h]: Do operators own?

File: src/include/optimizer/statistics/value_condition.h [hunk 4]
```diff
@@ -89,6 +89,6 @@ class ValueCondition {
   /**
    * Pointer to value stated in the condition
    */
-  std::shared_ptr<type::TransientValue> value_;
+  std::unique_ptr<type::TransientValue> value_;
```
Comment (lmwnshn) [inline src/include/optimizer/statistics/value_condition.h]: Does ValueCondition own their value?

File: src/network/postgres/postgres_protocol_interpreter.cpp [hunk 5]
```diff
@@ -112,7 +113,7 @@ bool PostgresProtocolInterpreter::TryReadPacketHeader(const std::shared_ptr<Read
   // Extend the buffer as needed
   if (curr_input_packet_.len_ > in->Capacity()) {
     // Allocate a larger buffer and copy bytes off from the I/O layer's buffer
-    curr_input_packet_.buf_ = std::make_shared<ReadBuffer>(curr_input_packet_.len_);
+    curr_input_packet_.buf_ = std::make_unique<ReadBuffer>(curr_input_packet_.len_);
     NETWORK_LOG_TRACE("Extended Buffer size required for packet of size {0}", curr_input_packet_.len_);
     curr_input_packet_.extended_ = true;
   } else {
```
Comment (lmwnshn) [inline src/network/postgres/postgres_protocol_interpreter.cpp]: So in the line below this one, that's when we currently force the shared_ptr usage. Can we get around this?

**Pull request merged**